entc/gen/type.go

```diff
@@ -680,21 +680,29 @@ func (t Type) RelatedTypes() []*Type {
 	return related
 }
 
-// check checks the schema type.
-func (t *Type) check() error {
-	pkg := t.Package()
-	if token.Lookup(pkg).IsKeyword() {
-		return fmt.Errorf("schema lowercase name conflicts with Go keyword %q", pkg)
+// CheckNameConflicts will determine if a name is going to conflict with any
+// pre-defined names
+func CheckNameConflicts(name string) error {
+	// Golang keywords and identifiers are lower case
+	lowerName := strings.ToLower(name)
+
```

Comment:
```suggestion
	// schema package is lower-cased (see Type.Package).
	pkg := strings.ToLower(name)
```

I prefer to keep it as `pkg` because it explains why this lower-cased identifier is needed.